File: app/src/main/java/com/dvinc/workouttimer/common/Assert.java
/*
 * Copyright (c) 2018 by Denis Verentsov (decsent@Yandex.ru)
 * All rights reserved.
 */

package com.dvinc.workouttimer.common;

import com.dvinc.workouttimer.BuildConfig;

public final class Assert {

    private Assert() {
    }

    public static <T> void assertNotNull(T object, String message) {
        if (!BuildConfig.DEBUG || object != null) {
            return;
        }
        throw new IllegalArgumentException(message);
    }

    public static <T> void assertNull(T object, String message) {
        if (!BuildConfig.DEBUG || object == null) {
            return;
        }
        throw new IllegalArgumentException(message);
    }
}

File: app/src/main/kotlin/com/dvinc/workouttimer/presentation/common/assert/Assert.kt
/*
 * Copyright (c) 2018 by Denis Verentsov (decsent@yandex.ru)
 * All rights reserved.
 */

package com.dvinc.workouttimer.presentation.common.assert

import com.dvinc.workouttimer.BuildConfig

@Deprecated("Delete after migration to viewModels")
class Assert {

    companion object {

        fun <T> assertNotNull(obj: T, message: String) {
            if (!BuildConfig.DEBUG || obj != null) {
                return
            }
            throw IllegalArgumentException(message)
        }

        fun <T> assertNull(obj: T, message: String) {
            if (!BuildConfig.DEBUG || obj == null) {
                return
            }
            throw IllegalArgumentException(message)
        }
    }
}
